packages/common/src/index.ts

```diff
@@ -56,6 +56,11 @@ export function arr<T>(length: number, mapper: (n: number) => T): T[] {
   return a;
 }
 
+/** @internal */
+export function slice<T>(length: number, index: number, values: T[]): T[] {
```

Comment:
I think this should have a less generic name, to differentiate it from `Array.slice()`. Maybe `sliceChunk()`?

Reply:
I think just `chunk()` sounds good. More or less aligns with the [Rust iterator](https://doc.rust-lang.org/std/primitive.slice.html#method.chunks).